prometheus/src/snapshot_metrics.rs

```diff
@@ -0,0 +1,65 @@
+use crate::utils::{write_metric, Metric, MetricFamily};
+use solana_runtime::snapshot_archive_info::{SnapshotArchiveInfo, SnapshotArchiveInfoGetter};
+use solana_runtime::snapshot_config::SnapshotConfig;
+use solana_runtime::snapshot_utils;
+use solana_sdk::clock::Slot;
+use std::io;
+
+pub fn write_snapshot_metrics<W: io::Write>(
+    snapshot_config: &Option<SnapshotConfig>,
+    out: &mut W,
+) -> io::Result<()> {
+    let snapshot_config = match snapshot_config.as_ref() {
```

Comment:
Not sure how to best structure the code in this function to avoid ton of indentations.

This seems pretty clear to me and is easy to follow/understand, though might not be most "idiomatic" Rust, so if you have some good idea how to change it happy to hear it.